File: 1-TwoSum/1-TwoSum.java
// Last updated: 8/13/2025, 8:24:50 PM
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int n = nums.length;
        HashMap<Integer,Integer> h = new HashMap();
        for(int i=0;i<n;i++){
            int num = nums[i];
            if(h.containsKey(target-num)){
                return new int[] {i,h.get(target-num)};
            }
            h.put(num,i);
        }
        return new int[] {};
    }
}

File: 1-TwoSum/1-TwoSum.cpp
// Last updated: 8/13/2025, 8:24:52 PM
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> ans;
        vector<int>::iterator i;
        vector<int>::iterator j;
        for(i=nums.begin();i<nums.end();i++)
        {   
            for(j=i+1;j<nums.end();j++){
                if(*i+*j==target){
                    ans.push_back(i - nums.begin());
                    ans.push_back(j - nums.begin());
                    //return ans;
                }
                else if(i==j)
                    //return ans;
                    break;
            }  
        }
        return ans;
    }
};
